File: smarttrade_app/lib/features/trading_journal/domain/repositories/journal_repository.dart
import 'package:dartz/dartz.dart';
import 'package:smarttrade_app/core/error/failures.dart';
import 'package:smarttrade_app/features/trading_journal/domain/entities/journal_entry.dart';

abstract class JournalRepository {
  Future<Either<Failure, List<JournalEntry>>> getEntries();

  Future<Either<Failure, Unit>> addEntry(JournalEntry entry);

  Future<Either<Failure, Unit>> deleteEntry(String id);
}

File: smarttrade-web/src/features/trading-journal/domain/repositories/journalRepository.ts
import type { Result } from '@/core/result/result'
import type { JournalEntry } from '../entities/journalEntry'

export interface JournalRepository {
  getEntries(): Promise<Result<JournalEntry[]>>
  addEntry(entry: JournalEntry): Promise<Result<void>>
  deleteEntry(id: string): Promise<Result<void>>
}
